examples/grid_example/public/app.tsx

```diff
@@ -196,7 +196,31 @@ export const GridExample = ({
             <EuiFlexItem grow={false}>
               <EuiFlexGroup gutterSize="s" alignItems="center">
                 <EuiFlexItem grow={false}>
-                  <AddEmbeddableButton pageApi={mockDashboardApi} uiActions={uiActions} />{' '}
+                  <AddEmbeddableButton pageApi={mockDashboardApi} uiActions={uiActions} />
+                </EuiFlexItem>
+                <EuiFlexItem grow={false}>
+                  <EuiButton
+                    onClick={() => {
+                      mockDashboardApi.rows$.next([
+                        ...mockDashboardApi.rows$.getValue(),
+                        {
+                          title: i18n.translate('examples.gridExample.defaultSectionTitle', {
+                            defaultMessage: 'New collapsible section',
+                          }),
+                          collapsed: false,
+                        },
+                      ]);
+                      // scroll to bottom after row is added
+                      new Promise((resolve) => setTimeout(resolve, 60)).then(() => {
+                        window.scrollTo({ top: document.body.scrollHeight, behavior: 'smooth' });
+                      });
```

Comment:
This works well enough as a "wait until row is rendered to scroll to the bottom of the screen" for the example app since, without this promise, the `scrollHeight` does not account for the new section when scrolling (we don't know exactly when the new row will mount).

In the real implementation on Dashboard, we could do a more robust and thorough implementation using a `MutationObserver` (see https://stackoverflow.com/a/61511955/28754956) but I'm leaning towards that being overkill even outside of the example app 🤷